src/app/icd/ICDManager.cpp

```diff
@@ -101,6 +107,56 @@ bool ICDManager::SupportsFeature(Feature feature)
 #endif // !CONFIG_BUILD_FOR_HOST_UNIT_TEST
 }
 
+void ICDManager::SendCheckInMsgs()
+{
+#if !CONFIG_BUILD_FOR_HOST_UNIT_TEST
+    VerifyOrDie(mStorage != nullptr);
+    VerifyOrDie(mFabricTable != nullptr);
+    for (const auto & fabricInfo : *mFabricTable)
+    {
+        uint16_t supported_clients = ICDManagementServer::GetInstance().GetClientsSupportedPerFabric();
+
+        ICDMonitoringTable table(*mStorage, fabricInfo.GetFabricIndex(), supported_clients /*Table entry limit*/,
+                                 mSymmetricKeystore);
+        if (!table.IsEmpty())
+        {
```

Comment:
is this extra check worth the optimization indent? Otherwise I would expect we could decrease method indent by just leaving the for loop. 

Or if needed something like:

```
if (table.IsEmpty())
{
   // avoid a loop over all items if all would return NOT_FOUND
   continue;
}
```